### PR TITLE
[Workspace] Refactor data source association panel

### PR DESCRIPTION
**Pull request by Kapian1234**
### Description
Refactor data source association panel

## Screenshot
before
Screenshot 2024-09-30 at 13 28 20

after
Screenshot 2024-09-30 at 16 17 34

## Changelog

- feat: Refactor data source association panel

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 1]
```diff
@@ -109,7 +117,7 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
           <EuiSpacer size="m" />
           {/* SelectDataSourcePanel is only visible for dashboard admin and when data source is enabled*/}
           {isDashboardAdmin && isDataSourceEnabled && (
-            <>
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:112]: Can we keep this fragment?

File: src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx [hunk 2]
```diff
@@ -130,7 +130,7 @@ export const SelectDataSourcePanel = ({
           <EuiFlexItem grow={false}>{addDirectQueryConnectionsButton}</EuiFlexItem>
         )}
       </EuiFlexGroup>
-      <EuiSpacer size="m" />
+      {assignedDataSourceConnections.length > 0 && <EuiSpacer size="m" />}
       <EuiFlexItem style={{ maxWidth: 768 }}>
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx:134]: Can we remove this FlexItem? Seems not flex container outside.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx:134]: But it is used to limit the width.

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 3]
```diff
@@ -133,10 +141,10 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
                 data-test-subj={`workspaceForm-dataSourcePanel`}
                 showDataSourceManagement={true}
               />
-              <EuiSpacer size="s" />
-              <EuiSpacer size="s" />
-            </>
+            </EuiPanel>
           )}
+          <EuiSpacer size="s" />
```
[the remark this answers is not in the document]
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:146]: Maybe these spacer could be deleted since we have padding in the panel now

File: src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx (hunk 2 above)
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx:134]: The max width already exists in the parent container. 
See https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx#L79
I think we can update like below:
```tsx
{assignedDataSourceConnections.length > 0 && <><EuiSpacer size="m" />{renderTableContent()}</>}
```

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:146]: oh, I get it. thank you

File: src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx (hunk 2 above)
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx:134]: I didn't notice that, thanks

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx (hunk 1 above)
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:112]: ok

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 4]
```diff
@@ -110,29 +118,31 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
...
             <>
-              <EuiTitle
-                {...generateRightSidebarScrollProps(RightSidebarScrollField.DataSource)}
-                size="s"
-              >
-                <h3>
-                  {i18n.translate('workspace.creator.form.associateDataSourceTitle', {
-                    defaultMessage: 'Associate data sources',
+              <EuiPanel>
+                <EuiTitle
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:122]: Can we use `EuiSelect + h2` as panel title [like this](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/workspace/public/components/workspace_creator/creator_details_panel.tsx#L42-L48) to align with the guidance.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:122]: ok

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 5]
```diff
@@ -110,29 +118,31 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
...
-              </EuiText>
-              <SelectDataSourcePanel
-                onChange={setSelectedDataSourceConnections}
-                savedObjects={savedObjects}
-                assignedDataSourceConnections={formData.selectedDataSourceConnections}
-                data-test-subj={`workspaceForm-dataSourcePanel`}
-                showDataSourceManagement={true}
-              />
+                </EuiText>
+                <SelectDataSourcePanel
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:138]: Do we need a EuiSpacer between description and SelectDataSourcePanel?

**Review by ruanyl — DISMISSED** [review]

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx (hunk 5 above)
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:138]: In SelectDataSourcePanel, there's already a `<EuiSpacer size="m" />`

File: src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx [hunk 6]
```diff
@@ -175,7 +174,7 @@ export const WorkspacePermissionSettingInput = ({
           placeholder={
             type === WorkspacePermissionItemType.User
               ? i18n.translate('workspaceForm.permissionSetting.selectUser', {
-                  defaultMessage: 'Enter user name or uer ID',
+                  defaultMessage: 'Enter user name or email address',
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx:177]: Currently we do not support email address, I think we should still use `user ID`.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx:177]: ok

File: src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx:1]: @Kapian1234 I think this file should not be included in this PR. Can we remove it?

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx:1]: removed

**Review by ruanyl — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx (hunk 6 above)
Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx]: ok I‘ll fix it in this PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8383/

**Review by wanglam — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]
